pkg/controller/istiocontrolplane/markings.go

```diff
@@ -0,0 +1,107 @@
+package istiocontrolplane
+
+import (
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"istio.io/operator/pkg/helmreconciler"
+)
+
+const (
+	// MetadataNamespace is the namespace for mesh metadata (labels, annotations)
+	MetadataNamespace = "install.operator.istio.io"
+
+	// OwnerNameKey represents the name of the owner to which the resource relates
+	OwnerNameKey = MetadataNamespace + "/owner-name"
+	// OwnerKindKey represents the kind of the owner to which the resource relates
+	OwnerKindKey = MetadataNamespace + "/owner-kind"
+	// OwnerGroupKey represents the group of the owner to which the resource relates
+	OwnerGroupKey = MetadataNamespace + "/owner-group"
+
+	// OwnerGenerationKey represents the generation to which the resource was last reconciled
+	OwnerGenerationKey = MetadataNamespace + "/owner-generation"
+)
+
+var (
+	// XXX: move this into a ConfigMap so users can override things if they add new types in customized charts
+	// ordered by which types should be deleted, first to last
```

Comment:
Seems like an overkill to allow users to add types through ConfigMap. If they go to such lengths, they probably forked Istio and should fork the operator too.

Reply:
yeah, i don't know.  this is really just for the pruning logic and there's probably not much else that could be added anyway.  i'll delete the comment.  thanks for the feedback.

Reply:
removed.